addons/website_sale/models/website_checkout_step.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.osv import expression
+
+
+class WebsiteCheckoutStep(models.Model):
+    _name = 'website.checkout.step'
+    _description = 'Website Checkout Step'
+    _inherit = ['website.published.multi.mixin']
+
+    name = fields.Char(required=True, translate=True)
+    sequence = fields.Integer()
+    step_href = fields.Char(string="Href", required=True)
+    main_button_label = fields.Char(translate=True)
+    back_button_label = fields.Char(translate=True)
+    active = fields.Boolean(default=True)
+    website_id = fields.Many2one('website', ondelete='cascade')
+
+    def _get_next_checkout_step(self, allowed_steps_domain):
+        """ Get the next step in the checkout flow based on the sequence."""
+
+        next_step_domain = expression.AND(
```

Comment:
There's a moment in `expression.py` that suggest to use `domain.py` instead.
>TODO deprecate and use Domain.AND(domains)
```suggestion
        next_step_domain = domain.AND(
```